server/game/cards/02_SHD/leaders/FennecShandHonoringTheDeal.ts

```diff
@@ -0,0 +1,51 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { LeaderUnitCard } from '../../../core/card/LeaderUnitCard';
+import { CardType, KeywordName, Location } from '../../../core/Constants';
+
+export default class FennecShandHonoringTheDeal extends LeaderUnitCard {
+    protected override getImplementationId() {
+        return {
+            id: '0911874487',
+            internalName: 'fennec-shand#honoring-the-deal',
+        };
+    }
+
+    protected override setupLeaderSideAbilities() {
+        this.addActionAbility({
+            title: 'Play a unit that costs 4 or less from your hand. Give it ambush for this phase',
+            cost: [AbilityHelper.costs.abilityResourceCost(1), AbilityHelper.costs.exhaustSelf()],
+            targetResolver: {
+                cardCondition: (card) => card.isUnit() && card.cost <= 4,
```

Comment:
Can remove the `isUnit()` from the card condition since you have a filter for it

Reply:
I can't, cost is not resolved if I deleted isUnit()